lib/elixir/lib/exception.ex

```diff
@@ -1342,8 +1342,21 @@ defmodule UndefinedFunctionError do
       hint_for_loaded_module(module, function, arity, nil)
   end
 
-  defp hint(_module, _function, _arity, _loaded?) do
-    ""
+  defp hint(module, function, arity, _loaded?) do
+    {module, _distance} =
+      Enum.map(:code.all_available(), fn {name, _, _} ->
+        {name, String.jaro_distance(to_string(module), to_string(name))}
+      end)
+      |> Enum.sort_by(&elem(&1, 1), :desc)
+      |> Enum.take(3)
+      |> Enum.find(fn {module, _distance} ->
+        module
+        |> to_string()
+        |> String.to_atom()
+        |> function_exported?(function, arity)
```

Comment:
Keep in mind `function_exported?` does not load the module. It may be fine to load the module if we do it after the threshold though. However, if we are specifically after casing, I would instead downcase the module names and see if there is a match.

Reply:
There's three cases I want to capture

1. The module is misspelled, e.g. `Enu` -> `Enum`
2. The module is miscased `.e.g. `enum` -> `Enum`
3. The module is not qualified `SomeModule` -> `Namespace.SomeModule`

Implementation-wise, I'm not sure that 1 and 3 are very different. #2 we can check by doing downcase and a string compare.

Can you help me understand the case where a module is *not* loaded, but it is the correct suggestion?

Reply:
A module may not be loaded at all. For example, if you do `mix compile`, `mix phx.server`, and type `PageHtml`, there is no guarantee the correct `PageHTML` will be loaded at all. It will only be loaded if it just been compiled or if something else had the correct name.

The other thing is that jaro distance doesn't consider casing, so we may have more false positives, especially because a large part of the module name will already be shared and taken into account by jaro (such as `MyLongAppNameWeb`). None of those issues happen with functions because the current module is by definition loaded and there is no namespace. In other words, it would be nice to have some real data, but it may be that we should take a different approach here.